### ISSUE TITLE
Bug: Pull Request failing testing kabanero 0.2.0 with CouldntGetResource

### ISSUE THREAD

**Issue by stephenkinder** [kabanero-io/kabanero-pipelines#108]
@csskinner25 commented on [Wed Oct 16 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105)

Describe the bug
Testing pipe line runs for kabanero 0.2.0. I have a webhook generated in tekton console.

I modified an app to cause a pipeline run to start.

In the tekton console the pipeline run fails with :

Unable to load PipelineRun details: CouldntGetResource
PipelineRun kabanero/web-mon-1571161246 can't be Run; it tries to bind Resources that don't exist: Couldn't retrieve PipelineResource: couldn't retrieve referenced input PipelineResource "web-mon-git-source-1571161246": pipelineresource.tekton.dev "web-mon-git-source-1571161246" not found

If I say rebuild in the tekton console for the failed run the request works and the app gets deployed.

Others have seen this problem also . I don't know if is a timing issue.
We also see the webhook get a 503 and if we redeliver the webhook gets created successfully also.
Not sure if these are related.

To Reproduce

    Install kabanero.
    Create webhook to github.ibm.com.
    Cause pull request to happen to create pipeline run.
    Pipeline fails sometimes with the resource not found problem.

Expected behavior
Should not be a failure in the run .

--

---

@gecock commented on [Wed Oct 16 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-542818347)

@tseelbach or @kvijai82 Could one of you take a look and work with Shelton to determine how serious this might be. Three testers have hit this issue in their recent testing.

---

@kvijai82 commented on [Wed Oct 16 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-542892678)

I discussed with Adam from the Tekton team a couple of weeks back and this was something they were not able to consistently reproduce.  It seemed like a timing issue where the pipelinerun did not wait for the resources to be created fully.  I thought I saw an update mentioned in the issue saying they might have put in a potential fix for this recently though.  We won't be getting the fix in 0.2 though.  I can add this to our pipeline troubleshooting guide.

---

@kvijai82 commented on [Wed Oct 16 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-542893095)

This is the issue in Tekton:  https://github.com/tektoncd/experimental/issues/240

---

@marikaj123 commented on [Wed Oct 16 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-542936000)

@gecock - you can remove the label release-0.2 if you are in agreement.

---

@gecock commented on [Thu Oct 17 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-543194787)

Yep I agree with Vijai. @kvijai82 Please add it to the troubleshooting guide for now until this gets addressed by Tekton and we can pull it in.

---

@gecock commented on [Thu Oct 17 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-543195217)

@kvijai82 could you post a link to the troubleshooting guide as well, please? 

---

@marikaj123 commented on [Thu Oct 17 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-543199775)

@tseelbach - I have added the labels to get us all to assess the issue with SVT and thereafter we can remove any unnecessary label.  SVT would like to get more info on this issue.

---

@kvijai82 commented on [Thu Oct 17 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-543211653)

@gecock Created a PR for the update to the doc here: https://github.com/kabanero-io/kabanero-pipelines/pull/92

---

@marikaj123 commented on [Mon Oct 28 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-547141320)

@kvijai82 - Is  tektoncd/experimental#240 a dependency required to be completed before we can complete this issue?

---

@anneqm commented on [Tue Oct 29 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-547622610)

 tektoncd/experimental#240 fix has a pre-requisite for Tekton pipeline 0.8.0 so will not be merged until after move to the newer version of the pipeline. 

---

@kvijai82 commented on [Fri Nov 01 2019](https://github.com/kabanero-io/kabanero-foundation/issues/105#issuecomment-548735060)

Moving it to next release when we can hopefully pick up tekton 0.8.

**Comment by marikaj123** [comment]
Waiting for Tekton 0.8 to be cut and then test it.

**Comment by marikaj123** [comment]
Tekton 0.8.0 has the fix and will be included in Kab v0.5.0.

**Comment by kaczyns** [comment]
@marikaj123 I am not aware of any commitment to pull Tekton 0.8.0 into Kab 0.5.0.  The pipelines team should comment on whether they think that this is feasible.

**Comment by marikaj123** [comment]
Vijai will talk to @aadeshpa to complete the work.

**Comment by anneqm** [comment]
This issue was before Triggers, so after move up to Tekton pipeline 0.8.0 this should be able to be closed. We believe tektoncd/experimental#240 is no longer needed with Triggers.

**Comment by anneqm** [comment]
@marikaj123 @kvijai82 @aadeshpa please let me know if you still see this issue after the move up to Tekton pipelines 0.8.0 with Tekton Triggers 0.1.0 and Tekton Dashboard and Webhooks Extension 0.3.0.

**Comment by aadeshpa** [comment]
@kvijai82 @anneqm : I verified by installing `Collections v0.4.0` in new OCP 4.2 cluster and then ran the below pipelinerun via webhooks and in all the cases the pipelinerun passed without any error. However this is without Tekton Triggers.

1. nodejs-express
2. nodejs
3. java-microprofile

**Comment by marikaj123** [comment]
Agreement with dev and SVT to close
